truss/templates/server/common/truss_server.py

```diff
@@ -154,6 +158,31 @@ def post(self, name: str):
         self.write(final_response)
 
 
+class LivenessAndLoadHandler(TrussHTTPHandler):
+
+    _load_thread: Thread = None
+
+    @staticmethod
+    def _exec_load(model):
+        model.load()
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def get(self):
+        for name in self.models:
+            model = self.models[name]
+
+            # if the model is ready or loading is in flight
+            if not model.ready and not self._load_thread:
+                self._load_thread = Thread(
+                    target=LivenessAndLoadHandler._exec_load, args=[model]
+                )
+                self._load_thread.start()
+
+            self.write("Alive")
+
+
 class TrussServer(KFServer):
```

Comment:
it might be worthwhile to override the `start` method here to assert that only one model is passed in and just keep it to one thread for now. We're forcing ourselves to follow the kfserving API which is making that loop above more complicated than it needs to be for one model.

Reply:
`inference_server.py` isn't really built to be called from within another python class, but there's no reason why i can't structure it to work that way. A good idea to not duplicate code there.